Orleans.Providers.MongoDB/ServiceCollectionExtensions.cs

```diff
@@ -16,5 +21,25 @@ public static IServiceCollection AddMongoDBClient(this IServiceCollection servic
 
             return services;
         }
+
+        /// <summary>
+        /// Configure silo to use MongoDb with a passed in connection string.
+        /// </summary>
+        /// <param name="services">The services.</param>
+        /// <param name="configureClientSettings">The configuration delegate to configure the <see cref="MongoClientSettings"/>.</param>
+        /// <returns></returns>
+        public static IServiceCollection AddMongoDBClient(this IServiceCollection services, ActionRef<MongoClientSettings> configureClientSettings)
+        {
+            if (configureClientSettings == null) 
+              throw new ArgumentNullException(nameof(configureClientSettings));
+            
+            var settings = new MongoClientSettings();
+            configureClientSettings(ref settings);
+
+            services.AddSingleton<IMongoClient>(c => new MongoClient(settings));
+            services.AddSingleton<IMongoClientFactory, DefaultMongoClientFactory>();
```

Comment:
I think calls to `services.AddSingleton<IMongoClientFactory, DefaultMongoClientFactory>();` need to be updated to `TryAddSingleton` including in the other overload. Allowing users to set the default themselves.